enclave_wrangler/dataset_upload.py

```diff
@@ -34,11 +35,131 @@
 CSET_UPLOAD_REGISTRY_PATH = os.path.join(UPLOADS_DIR, 'cset_upload_registry.csv')
 
 
-def post_to_enclave_and_update_code_sets_csv(input_csv_folder_path) -> pd.DataFrame:
+# TODO: Rewrite to use new API
+def post_to_enclave(input_csv_folder_path: str, enclave_api=['old_rid_based', 'default'][1]):
     """Uploads data to enclave and updates the following column in the input's code_sets.csv:
+    # TODO: Siggie: updating these files is probably not worth / necessary right now
     - enclave_codeset_id
     - enclave_codeset_id_updated_at
-    - concept_set_name"""
+    - concept_set_name
+
+    :param enclave_api (str): One of 'old_rid_based' or 'default'. The 'old_rid_based' Enclave API is the one which
+    required RIDs for params rather than the param name, and there may be other differences. 'default' refers to the
+    newer Enclave API (as of 2022/10).
+    """
+    if enclave_api == 'old_rid_based':
+        return post_to_enclave_old_api(input_csv_folder_path)
+    if DEBUG:
+        log_debug_info()
+
+    # Read data
+    concept_set_container_edited_df = pd.read_csv(
+        os.path.join(input_csv_folder_path, 'concept_set_container_edited.csv')).fillna('')
+    concept_set_container_edited_d = json.loads(concept_set_container_edited_df.to_json(orient='records'))
+    code_sets_df = _load_standardized_input_df(os.path.join(input_csv_folder_path, 'code_sets.csv'))  # .fillna('')
+    concept_set_version_item_rv_edited_df = pd.read_csv(
+        os.path.join(input_csv_folder_path, 'concept_set_version_item_rv_edited.csv')).fillna('')
+
+    # Link data into single iterable
+    # todo: @Siggie: You mentioned to delete the internal_id we assigned, but I don't know how to link these togehter otherwise
+    linked_csets: Dict = {}
+    for container_d in concept_set_container_edited_d:
+        cset_name = container_d['concept_set_id']
+        linked_csets[cset_name] = {'versions': {}}
+        linked_csets[cset_name]['container'] = container_d
+        code_sets_df_i = code_sets_df[code_sets_df['concept_set_name'] == cset_name]
+        code_sets_d_i = json.loads(code_sets_df_i.to_json(orient='records'))
+        for version_d in code_sets_d_i:
+            version_id = version_d['codeset_id']
+            linked_csets[cset_name]['versions'][version_id] = {'items': []}
+            linked_csets[cset_name]['versions'][version_id]['version'] = version_d
+            concept_set_version_item_rv_edited_df_i = concept_set_version_item_rv_edited_df[
+                concept_set_version_item_rv_edited_df['codeset_id'] == version_id]
+            concept_set_version_item_rv_edited_d_i = json.loads(concept_set_version_item_rv_edited_df_i.to_json(orient='records'))
+            for item_d in concept_set_version_item_rv_edited_d_i:
+                linked_csets[cset_name]['versions'][version_id]['items'].append(item_d)
+
+    # Upload to enclave
```

Comment:
@Sigfried There are some params I was not able to fill out. For some of them, I've added comments to this PR. But for others, I've simply left comments in the code.